### DISCUSSION TITLE
Air-gapped use of nuclei

### DISCUSSION THREAD

**Discussion by smijolovic** [projectdiscovery/nuclei#1607] [Q&A] [answered]
I've built and packaged nuclei and the nuclei-templates, but can not get nuclei to use the local templates.

The templates are installed in /usr/local/nuclei-templates.

When I run the command: nuclei -t /usr/local/nuclei-templates/ 

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] nuclei-templates are not installed, installing...
[INF] Successfully downloaded nuclei-templates (v8.8.4) to /home/chaasmadmin/nuclei-templates. GoodLuck!
[INF] Using Nuclei Engine 2.6.0 (latest)
[INF] Using Nuclei Templates 8.8.4 (latest)
[INF] Templates added in last update: 3030
[INF] Templates loaded for scan: 2823

How do you specify the templates directory?

**Comment by forgedhallpass** [comment]
The template directory configuration is stored under `~/.config/nuclei/.templates-config.json` and `~/.config/nuclei/config.yaml`.

You can use the `-ud` or `-update-directory` command line arguments to change the default template location. See `nuclei -help`

**Reply by smijolovic** [reply]
That's still not working. The templates are already there:

chaasmadmin@maestro-dogfood-source:/usr/local/nuclei-templates$ ls
CODE_OF_CONDUCT.md TEMPLATES-STATS.json cves file miscellaneous technologies
CONTRIBUTING.md TEMPLATES-STATS.md default-logins fuzzing misconfiguration token-spray
LICENSE.md TOP-10.md dns headless network vulnerabilities
PULL_REQUEST_TEMPLATE.md cnvd exposed-panels helpers ssl workflows
README.md contributors.json exposures iot takeovers

When I run: nuclei -ud /usr/local/nuclei-templates/

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] nuclei-templates are not installed, installing...
[ERR] Could not update templates: failed to write templates: could not create template file: open /usr/local/nuclei-templates/CODE_OF_CONDUCT.md: permission denied

**Comment by smijolovic** [comment, marked as answer]
I figured it out!

nuclei -ud /usr/local/nuclei-templates -duc

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.6.0 (latest)
[INF] Using Nuclei Templates  (outdated)
[INF] Templates added in last update: 0
[INF] Templates loaded for scan: 2823

Wonder why it says outdated when I have packaged the latest 8.8.4 templates.
